packages/concerto-tools/lib/codegen/fromJsonSchema/cto/jsonSchemaVisitor.js

```diff
@@ -0,0 +1,958 @@
+                        ...(
+                            parameters.pathToDefinitions ||
+                                ['definitions']
+                        ),
+                        definitionName
+                    ],
+                );
+
+                return (
+                    new Definition(definition, pathToDefinition)
+                ).accept(
+                    this, {
+                        ...parameters, traversedReferences: [
+                            ...traversedReferences,
+                            reference.body
+                        ]
+                    }
+                );
+            }
+        }
+
+        return { name: this.inferInlineObjectConceptName(pathToDefinition) };
+    }
+    /**
+     * Reference property visitor.
+     * @param {Object} reference - a JSON Schema model reference property.
+     * @param {Object} parameters - the visitor parameters.
+     *
+     * @return {Object} the Concerto fields and declarations inferred from
+     * the JSON Schema model reference property. A property can spawn
+     * declarations as well, if it contains inline objects.
+     * @private
+     */
+    visitReference(reference, parameters) {
+        if (this.isStringLocalReference(reference.body)) {
+            return (new LocalReference(reference.body, reference.path))
+                .accept(this, parameters);
+        }
+        // TODO: Handle remote reference.
+        // TODO: Handle URL reference.
+    }
+    /**
+     * Array property visitor.
+     * @param {Object} arrayProperty - a JSON Schema model array property.
+     * @param {Object} parameters - the visitor parameters.
+     *
+     * @return {Object} the Concerto fields and declarations inferred from
+     * the JSON Schema model array property. A property can spawn declarations
+     * as well, if it contains inline objects.
+     * @private
+     */
+    visitArrayProperty(arrayProperty, parameters) {
+        parameters.assignableFields = {
+            ...parameters.assignableFields,
+            isArray: true,
+        };
+
+        return (new Property(arrayProperty.body.items, arrayProperty.path))
+            .accept(this, parameters);
+    }
+    /**
+     * Property visitor.
+     * @param {Object} property - a JSON Schema model property.
+     * @param {Object} parameters - the visitor parameters.
+     *
+     * @return {Object} the Concerto fields and declarations inferred from
+     * the JSON Schema model property. A property can spawn declarations as
+     * well, if it contains inline objects.
+     * @private
+     */
+    visitProperty(property, parameters) {
+        const propertyName = property.path[
+            property.path.length - 1
+        ];
+
+        const propertyProperties = {
+            ...{
+                name: propertyName,
+                isArray: false,
+                isOptional: !parameters.required
+                    ?.includes(propertyName),
+            },
```

Comment:
```suggestion
            name: propertyName,
            isArray: false,
            isOptional: !parameters.required
                    ?.includes(propertyName),
```

Why do you need this spread here?